src/messaging/ExchangeMgr.h

```diff
@@ -183,6 +183,10 @@ class DLL_EXPORT ExchangeManager : public SessionMessageDelegate
      */
     void CloseAllContextsForDelegate(const ExchangeDelegate * delegate);
 
+    // This API is used by UpdateNOC command, to abort all exchanges except the given one, whose abort is deferred until UpdateNOC
+    // command finishing its work.
```

Comment:
```suggestion
    // This API is used by commands that need to shut down all existing exchanges on 
    // a fabric but need to make sure the response to the command still goes out on the exchange
    // the command came in on.  This API flags that one exchange to shut down its session
    // when it's done.
```
As followup we should make RemoveFabric use this too, instead of its current delegate hack, right?